zerver/tests/test_docs.py

```diff
@@ -58,12 +58,30 @@ def _check_basic_fetch(
         url: str,
         subdomain: str,
         expected_strings: Sequence[str],
+        allow_robots: bool,
     ) -> "TestHttpResponse":
+        # For whatever reason, we have some urls that don't follow
+        # the same policies as the majority of our urls.
+        if url.startswith("/integrations/doc-html"):
+            allow_robots = True
+
+        if url.startswith("/attribution/"):
+            allow_robots = False
```

Comment:
For this commit, I think I preferred the previous model because this bit of configuration feels more readable where it was written.

BTW for /integrations/doc-html/, it's because those are essentially API URLs to fetch the content for /integrations pages.

Reply:
I split out a separate function for this and added a comment.